File: src/main/java/com/qtech/bubbles/core/utils/categories/ShapeUtils.java
package com.qtech.bubbles.core.utils.categories;

import java.awt.*;
import java.awt.geom.Area;

public class ShapeUtils {
    public static boolean checkIntersection(Shape shapeA, Shape shapeB) {
        Area areaA = new Area(shapeA);
        areaA.intersect(new Area(shapeB));
        return !areaA.isEmpty();
    }
}

File: src/main/kotlin/qtech/bubbles/core/utils/categories/ShapeUtils.kt
package qtech.bubbles.core.utils.categories

import java.awt.Shape
import java.awt.geom.Area

object ShapeUtils {
    @JvmStatic
    fun checkIntersection(shapeA: Shape?, shapeB: Shape?): Boolean {
        val areaA = Area(shapeA)
        areaA.intersect(Area(shapeB))
        return !areaA.isEmpty
    }
}